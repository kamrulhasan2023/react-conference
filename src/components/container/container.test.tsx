import { render } from '@testing-library/react';
import Container from '.';

it('renders container correctly', () => {
  const { getByRole } = render(
    <Container>
      <h2>Container</h2>
    </Container>
  );
  const element = getByRole('presentation');
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

describe.each([true, false])('Given container isFluid: %s', (isFluid) => {
  it(`Expect container to have class ${isFluid}`, () => {
    const { getByRole } = render(
      <Container isFluid={isFluid}>
        <h2>Container</h2>
      </Container>
    );
    const element = getByRole('presentation');
    if (isFluid) {
      expect(element).toHaveClass('full-width');
    } else {
      expect(element).not.toHaveClass('full-width');
    }
  });
});

describe.each([true, false])(
  'Given container isNoPadding: %s',
  (isNoPadding) => {
    it(`Expect container to have class ${isNoPadding}`, () => {
      const { getByRole } = render(
        <Container isNoPadding={isNoPadding}>
          <h2>Container</h2>
        </Container>
      );
      const element = getByRole('presentation');
      if (isNoPadding) {
        expect(element).toHaveClass('no-padding');
      } else {
        expect(element).not.toHaveClass('no-padding');
      }
    });
  }
);
