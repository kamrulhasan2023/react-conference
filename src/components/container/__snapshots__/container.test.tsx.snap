// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders container correctly 1`] = `
<div
  class="container"
  role="presentation"
>
  <h2>
    Container
  </h2>
</div>
`;
